src/languages/nix.js

```diff
@@ -9,85 +9,319 @@ Category: system
+      "bitAnd",
+      "bitOr",
+      "bitXor",
+      "break",
       "builtins",
+      "catAttrs",
+      "ceil",
+      "compareVersions",
+      "concatLists",
+      "concatMap",
+      "concatStringsSep",
+      "convertHash",
+      "currentSystem",
+      "currentTime",
+      "deepSeq",
+      "derivation",
+      "derivationStrict",
+      "dirOf",
+      "div",
+      "elem",
+      "elemAt",
+      "false",
+      "fetchGit",
+      "fetchMercurial",
+      "fetchTarball",
+      "fetchTree",
+      "fetchurl",
+      "filter",
+      "filterSource",
+      "findFile",
+      "flakeRefToString",
+      "floor",
+      "foldl'",
+      "fromJSON",
+      "fromTOML",
+      "functionArgs",
+      "genList",
+      "genericClosure",
+      "getAttr",
+      "getContext",
+      "getEnv",
+      "getFlake",
+      "groupBy",
+      "hasAttr",
+      "hasContext",
+      "hashFile",
+      "hashString",
+      "head",
+      "import",
+      "intersectAttrs",
+      "isAttrs",
+      "isBool",
+      "isFloat",
+      "isFunction",
+      "isInt",
+      "isList",
+      "isNull",
+      "isPath",
+      "isString",
+      "langVersion",
+      "length",
+      "lessThan",
+      "listToAttrs",
       "map",
+      "mapAttrs",
+      "match",
+      "mul",
+      "nixPath",
+      "nixVersion",
+      "null",
+      "parseDrvName",
+      "parseFlakeRef",
+      "partition",
+      "path",
+      "pathExists",
+      "placeholder",
+      "readDir",
+      "readFile",
+      "readFileType",
       "removeAttrs",
+      "replaceStrings",
+      "scopedImport",
+      "seq",
+      "sort",
+      "split",
+      "splitVersion",
+      "storeDir",
+      "storePath",
+      "stringLength",
+      "sub",
+      "substring",
+      "tail",
       "throw",
+      "toFile",
+      "toJSON",
+      "toPath",
       "toString",
-      "derivation"
-    ]
+      "toXML",
+      "trace",
+      "traceVerbose",
+      "true",
+      "tryEval",
+      "typeOf",
+      "unsafeDiscardOutputDependency",
+      "unsafeDiscardStringContext",
+      "unsafeGetAttrPos",
+      "warn",
+      "zipAttrsWith",
+    ].map(b => `builtins\\.${b}`)),
   };
-  const ANTIQUOTE = {
-    className: 'subst',
-    begin: /\$\{/,
-    end: /\}/,
-    keywords: KEYWORDS
+
+  const IDENTIFIER_REGEX = '[A-Za-z_][A-Za-z0-9_\'-]*';
+
+  const LOOKUP_PATH = {
+    scope: 'symbol',
+    match: new RegExp(`<${IDENTIFIER_REGEX}(/${IDENTIFIER_REGEX})*>`),
   };
-  const ESCAPED_DOLLAR = {
-    className: 'char.escape',
-    begin: /''\$/,
+
+  const PATH_PIECE = "[A-Za-z0-9_\\+\\.-]+"
+  const PATH = {
+    scope: 'symbol',
+    match: new RegExp(`(\\.\\.|\\.|~)?/(${PATH_PIECE})?(/${PATH_PIECE})*(?=[\\s;])`),
   };
+
+  const OPERATOR = {
+    scope: 'operator',
+    relevance: 0,
+    match: hljs.regex.either(...[
+      '==',
+      '=',
+      '\\+\\+',
+      '\\+',
+      '<=',
+      '<\\|',
+      '<',
+      '>=',
+      '>',
+      '->',
+      /(?<=(\s|[0-9\.]+))-/,
```

Comment:
Sadly, we can't use look-behind until version 12. :-/

Reply:
Hmm, I might need some time to think about a workaround for this one